cms/djangoapps/contentstore/rest_api/v1/views/videos.py

```diff
@@ -32,6 +46,162 @@
+            encodings_download_url: '/video_encodings_download/course_id',
+            default_video_image_url: '/static/studio/images/video-images/default_video_image.png',
+            previous_uploads: [
+                {
+                    edx_video_id: 'mOckID1',
+                    clientVideoId: 'mOckID1.mp4',
+                    created: '',
+                    courseVideoImageUrl: '/video',
+                    transcripts: [],
+                    status: 'Imported',
+                    file_size: 123,
+                    download_link: 'http:/download_video.com'
+                },
+                {
+                    edx_video_id: 'mOckID5',
+                    clientVideoId: 'mOckID5.mp4',
+                    created: '',
+                    courseVideoImageUrl: 'http:/video',
+                    transcripts: ['en'],
+                    status: 'Failed',
+                    file_size: 0,
+                    download_link: ''
+                },
+                {
+                    edx_video_id: 'mOckID3',
+                    clientVideoId: 'mOckID3.mp4',
+                    created: '',
+                    courseVideoImageUrl: null,
+                    transcripts: ['en'],
+                    status: 'Ready',
+                    file_size: 123,
+                    download_link: 'http:/download_video.com'
+                },
+            ],
+            concurrent_upload_limit: 4,
+            video_supported_file_formats: ['.mp4', '.mov'],
+            video_upload_max_file_size: '5',
+            video_image_settings: {
+                video_image_upload_enabled: false,
+                max_size: 2097152,
+                min_size: 2048,
+                max_width: 1280,
+                max_height: 720,
+                supported_file_formats: {
+                    '.bmp': 'image/bmp',
+                    '.bmp2': 'image/x-ms-bmp',
+                    '.gif': 'image/gif',
+                    '.jpg': 'image/jpeg',
+                    '.jpeg': 'image/jpeg',
+                    '.png': 'image/png',
+                },
+            },
+            is_video_transcript_enabled: false,
+            active_transcript_preferences: null,
+            transcript_credentials: {},
+            transcript_available_languages: [{ language_code: 'ab', language_text: 'Abkhazian' }],
+            video_transcript_settings: {
+                transcript_download_handler_url: '/transcript_download/',
+                transcript_upload_handler_url: '/transcript_upload/',
+                transcript_delete_handler_url: '/transcript_delete/course_id',
+                trancript_download_file_format: 'srt',
+            },
+            pagination_context: {},
+        }
+        ```
+        """
+        course_key = CourseKey.from_string(course_id)
+
+        if not has_studio_read_access(request.user, course_key):
+            self.permission_denied(request)
+
+        with modulestore().bulk_operations(course_key):
```

Comment:
So... I still believe that this rest api view should not call modulestore anywhere, because it should be agnostic for what is done with the data or how it is retrieved. Can you extract line 151 to 156 to a function and put that function into video_storage_handlers?